shm/Island.hpp

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include "Coordinates.hpp"
+
+class Island {
+private:
+    Coordinates position_;
+public:
+    Island(size_t positionX, size_t positionY) : position_(positionX, positionY) {};
+    const Coordinates& getPosition() { return position_; };
```

Comment:
Is it better to return const& or copy of Coordinates in this case?

Reply:
@PierwszyKrytykRP 
1) Is it better to return const& or copy of Coordinates in this case?

2) And why not `const Coordinates& getPosition() const { return position_; };`

**`const` after the function signature** will ensure us that this function won't modify class member values.

In your code I can do this:
```
const Coordinates& getPosition() { 
      Coordinates tempPos(9,9);
      position_ = tempPos;
      return position_; 
};
``` 
And it's totally fine.

Reply:
I forgot about const after signature. I will change that
And I think if you can do that with this code you could also remove const and still do this :D 

AFAIK reference of Coordinates class is smaller than Coordinates object itself so it more efficient to return reference

Reply:
For sure reference of Coordinates class is smaller than Coordinates object itself and it's more efficient. I'm just wondering about lifetime of these objects. Do you think it would be possible to return Coordinates& (reference to position_ which is part of Island object), store it somewhere, destroy the Island object and... what will happen with this reference to position_? 

Just asking - probably we should not care about it for now.